rcl/src/rcl/arguments.c

```diff
@@ -51,6 +51,53 @@ _rcl_parse_remap_rule(
   rcl_allocator_t allocator,
   rcl_remap_t * output_rule);
 
+RCL_LOCAL
+rcl_ret_t
+_rcl_parse_param_rule(
+  const char * arg,
+  rcl_allocator_t allocator,
+  char ** output_rule);
+
+rcl_ret_t
+rcl_arguments_get_param_files(
+  const rcl_arguments_t * arguments,
+  rcl_allocator_t allocator,
+  char *** parameter_files)
+{
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(arguments, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(arguments->impl, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(parameter_files, RCL_RET_INVALID_ARGUMENT, allocator);
+  *(parameter_files) = allocator.allocate(
+    sizeof(char *) * arguments->impl->num_param_files_args, allocator.state);
+  if (NULL == *parameter_files) {
+    return RCL_RET_BAD_ALLOC;
+  }
+  for (int i = 0; i < arguments->impl->num_param_files_args; ++i) {
+    (*parameter_files)[i] = rcutils_strdup(arguments->impl->parameter_files[i], allocator);
+    if (NULL == *parameter_files) {
+      // deallocate allocated memory
+      for (int r = i; r >= 0; --r) {
+        allocator.deallocate((*parameter_files[i]), allocator.state);
+      }
+      allocator.deallocate((*parameter_files), allocator.state);
+      (*parameter_files) = NULL;
+      return RCL_RET_BAD_ALLOC;
+    }
+  }
+  return RCL_RET_OK;
+}
+
+int
+rcl_arguments_get_param_files_count(
+  const rcl_arguments_t * args)
```

Comment:
nitpick: Arguments don't need to be wrapped here.

Reply:
Same for declaration in header. Those I suppose you're just trying to always wrap or not, so maybe this is fine, just looked weird at a glance.

Reply:
fair point. TBH I didn't look twice and copied an existing function from that file.
https://github.com/ros2/rcl/blob/18a6ef780bab325915ba522df401d855e88489c0/rcl/include/rcl/arguments.h#L103-L105

 It may be worth a follow up to clean it in the entire file.